components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,496 @@
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  resourceProfiles:
+    function: S / M / L / XL / ... | (empty)-> resources field has to be filled
+    build: S / M / L / XL / ... | (empty)-> resources field has to be filled
+  replicas: 1
+  scalingConfig:
+    min: 1
+    max: 2
+  templates:
+    functionPod:
+      metadata: # labels and annotations only for function pod
+        labels: 
+          app: my-app
+        annotations: 
+          fluentbit.io/parser: my-regex-parser
+          istio-injection: enabled
+      spec: 
+        resources: # optional... required if spec.resourceProfiles.function is empty
+          limits: ... #if_custom
+          requests: ... #if_custom
+        env: # function pod envs
+          - name: PASSWORD
+            valueFrom:
+              secretRef:
+                name: mysvc-passwords
+                key: password
+          - name: EXTERNAL_API_URL
+            valueFrom:
+              configmapRef:
+                name: mysvc-configuration
+                key: URL
+        volumeMounts: ...
+          - name: config
+            mountPath: /etc/config/
+          - name: search-index
+            mountPath: /etc/index
+      volumes:
+        - name: search-index
+            nfs:
+              path: /path-to-index
+              readOnly: true
+              server: localhost
+        - name: config
+          configmap:
+            name: function-configuration
+            items:
+              - key: config
+                path: config.yaml
+    buildPod:
+      metadata: ...# labels and annotations only for build pod
+      spec: 
+        resources: # optional... required if spec.resourceProfiles.build is empty
+          limits: ... #if_custom
+          requests: ... #if_custom
+        env: ...
+        volumeMounts: ...
+      volumes: ...
+```
+
+
+### Compromise
+
+ - clearly separate configuration of the build stage in the spec.
+ - add simple, [purpose focused](https://blogs.sap.com/2022/07/12/the-new-way-to-consume-service-bindings-on-kyma-runtime/) `serviceBindings` 
+ - volume mounts separated to a different place, for more advanced case (implemented once requested) 
+ - don't group labels and annotations under metadata. 
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  replicas: 1
+  scalingConfig:
+    min: 1
+    max: 2
+
+  labels: #?
+    app: my-app
+  annotations: #?
+    fluentbit.io/parser: my-regex-parser
+    istio-injection: enabled
+
+  serviceBindings: 
+  - secretName: my-secret
+    mountPath: "/foo" # optional.. read from SERVICE_BINDING_ROOT ENV 
+
+  profile: S / M / L / XL / ... / Custom #optional
```

Comment:
I wouldn't use `Custom` to enable manual resoruces. The simplier to use option is to leave it empty.

Reply:
and fail validation if both are specified

Reply:
included in the PR